gulpfile.js

```diff
@@ -1,57 +1,135 @@
-    sourcemap   = require('gulp-sourcemaps'),
-    runSequence = require('run-sequence'),
-    readme      = require('gulp-readme-to-markdown');
-
-var config = {
+const fs           = require('fs');
+const browserSync  = require('browser-sync').create();
+const gulp         = require('gulp');
+const include      = require('gulp-include');
+const eslint       = require('gulp-eslint');
+const isFixed      = require('gulp-eslint-if-fixed');
+const babel        = require('gulp-babel');
+const rename       = require('gulp-rename');
+const uglify       = require('gulp-uglify');
+const readme       = require('gulp-readme-to-markdown');
+const merge        = require('merge');
+
+
+let config = {
   src: {
-    js: './src/js'
+    jsPath: './src/js'
   },
   dist: {
-    js: './static/js'
-  }
+    jsPath: './static/js'
+  },
+  packagesPath: './node_modules',
+  sync: false,
+  syncTarget: 'http://localhost/wordpress/'
 };
 
-gulp.task('es-lint', function() {
-  return gulp.src(config.src.js + '/**/*.js')
-    .pipe(eslint({fix: true}))
+/* eslint-disable no-sync */
+if (fs.existsSync('./gulp-config.json')) {
+  const overrides = JSON.parse(fs.readFileSync('./gulp-config.json'));
+  config = merge(config, overrides);
+}
+/* eslint-enable no-sync */
+
+
+//
+// Helper functions
+//
+
+// Base JS linting function (with eslint). Fixes problems in-place.
+function lintJS(src, dest) {
+  dest = dest || config.src.jsPath;
+
+  return gulp.src(src)
+    .pipe(eslint({
+      fix: true
+    }))
     .pipe(eslint.format())
-    .pipe(isFixed(config.src.js));
-});
+    .pipe(isFixed(dest));
+}
 
-gulp.task('js-build', function() {
-  return gulp.src(config.src.js + '/script.js')
+// Base JS compile function
+function buildJS(src, dest) {
+  dest = dest || config.dist.jsPath;
+
+  return gulp.src(src)
     .pipe(include({
-      includePaths: [config.src.js]
+      includePaths: [config.packagesPath, config.src.jsPath]
     }))
-    .on('error', console.log)
+    .on('error', console.log) // eslint-disable-line no-console
     .pipe(babel())
     .pipe(uglify())
-    .pipe(rename('ucf-page-assets.min.js'))
-    .pipe(gulp.dest(config.dist.js));
+    .pipe(rename({
+      extname: '.min.js'
+    }))
+    .pipe(gulp.dest(dest));
+}
+
+// BrowserSync reload function
+function serverReload(done) {
+  if (config.sync) {
+    browserSync.reload();
+  }
+  done();
+}
+
+// BrowserSync serve function
+function serverServe(done) {
+  if (config.sync) {
+    browserSync.init({
+      proxy: {
+        target: config.syncTarget
+      }
+    });
+  }
+  done();
+}
+
+
+//
+// JavaScript
+//
+
+// Run eslint on js files in src.jsPath
+gulp.task('es-lint-plugin', () => {
+  return lintJS([`${config.src.jsPath}/*.js`], config.src.jsPath);
 });
 
-gulp.task('js', function() {
-  runSequence('es-lint', 'js-build');
+// Concat and uglify js files through babel
+gulp.task('js-build-plugin', () => {
+  return buildJS(`${config.src.jsPath}/script.js`, config.dist.jsPath);
```

Comment:
The main js file for the plugin will have its name changed with this update, since `buildJS()` uses the filename of the source file with `.min.js` appended to the end.  You can see in this PR that there are now two minified JS files in the project--script.min.js and ucf-page-assets.min.js.

You could either delete the old ucf-page-assets.min.js file and stick with the new script.min.js file moving forward, or rename the source script.js file to ucf-page-assets.js to maintain the same minified file name.

Reply:
good catch